src/main/scala/com/gu/newsletterlistcleanse/services/Newsletters.scala

```diff
@@ -32,11 +30,19 @@ class Newsletters {
 
   implicit val sttpBackend: SttpBackend[Future, Nothing, WebSocketHandler] = SttpFactory.createSttpBackend()
 
-  def fetchAllNewsletters(): EitherT[Future, String, List[String]] = {
-    def parseBody(bodyString: String): Either[String, List[String]] = {
+  def filterNewsletters(allNewsletters: List[Newsletter], filterList: List[String] = Nil): List[Newsletter] =
+    filterList match {
+      case Nil =>
+        allNewsletters
+      case _ =>
+        allNewsletters.filter(newsletter => filterList.contains(newsletter.brazeNewsletterName))
+    }
```

Comment:
This works and does exactly what we want so I don't think we should change it, but here's another way to express the same thing
```suggestion
    allNewsletters.filter(newsletter => filterList.isEmpty || filterList.contains(newsletter.brazeNewsletterName))
```

Reply:
That's good to know! Thanks for the tip